status: 2023q3: mfsBSD building integration (#252)
freebsd/freebsd-doc: website/content/en/status/report-2023-07-2023-09/mfsbsd-build-integration.adoc

> === Integrate mfsBSD into the release building tools

Reviewer: ```suggestion
=== Integrate mfsBSD into the Release Building Tools
```

Changes of case to Americanise headings? 

For consistency within e.g. the table of contents of the compiled report. 

https://docs.freebsd.org/en/books/fdp-primer/book/#writing-style-guidelines

Links: +

> link:https://wiki.freebsd.org/SummerOfCode2023Projects/IntegrateMfsBSDIntoTheReleaseBuildingTools[Wiki Article] URL: link:https://wiki.freebsd.org/SummerOfCode2023Projects/IntegrateMfsBSDIntoTheReleaseBuildingTools[]
> link:https://reviews.freebsd.org/D41705[Code Review on Phabricator] URL: link:https://reviews.freebsd.org/D41705[]

Reviewer: ```suggestion
link:https://wiki.freebsd.org/SummerOfCode2023Projects/IntegrateMfsBSDIntoTheReleaseBuildingTools[Wiki Article] URL: link:https://wiki.freebsd.org/SummerOfCode2023Projects/IntegrateMfsBSDIntoTheReleaseBuildingTools[]+
link:https://reviews.freebsd.org/D41705[Code Review on Phabricator] URL: link:https://reviews.freebsd.org/D41705[]+
```

link:https://freebsdfoundation.org/blog/meet-the-2023-freebsd-google-summer-of-code-students-soobin-rho/[FreeBSD Foundation Blog Post] URL: link:https://freebsdfoundation.org/blog/meet-the-2023-freebsd-google-summer-of-code-students-soobin-rho/[]

Contact: Soobin Rho <soobinrho@FreeBSD.org>

==== What is mfsBSD?

> "mfsBSD is a toolset to create small-sized but full-featured mfsroot based distributions of FreeBSD that store all files in memory (MFS) [Memory File System] and load from hard drive, usb storage device or optical media.

Reviewer: ```suggestion
"mfsBSD is a toolset to create small-sized but full-featured mfsroot based distributions of FreeBSD that store all files in memory (MFS) [Memory File System] and load from hard drive, USB storage device, or optical medium.
```

(singular, singular, singular)

> "mfsBSD is a toolset to create small-sized but full-featured mfsroot based distributions of FreeBSD that store all files in memory (MFS) [Memory File System] and load from hard drive, USB storage device, or optical medium.

Reviewer 2: "Optical medium" seems wrong here. It is pedanticly correct, but obscure. 
I'd be tempted to use "cdrom" instead. It's more approachable and accurate enough to convey the meaning without delving down into the obscure.

As a side note: why not a unified image for both iso and img? Most other projects have long since gone that way and we have at least some support for that in our tree.

Author: Thank you, @bsdimp!

- Good point! However, I'd be wary of changing the wording here because it's a quote from the [mfsBSD white paper](https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf).
- This mfsBSD integration project just imports mfsBSD into the FreeBSD release tools, and I don't actually have control over the upstream mfsBSD source code. Nice idea nonetheless. Thanks guys 👍

Reviewer 2: Ok on the quote.
There's an issue that I have raised in the code review though that might require changes to mfsbsd...

Author: Hey @bsdimp, I'll definitely research about the unified image format in mfsBSD. Thank you for pointing me to the right direction. It will be very helpful! Making changes to the code revision now.

Reviewer 2: You bet. I also understand, btw, GSoC timelines and scopes of project. I'm not looking to make things difficult for you if my questions/comments lead to something unreasonable... but I'm happy to help make things better.

It can be used for a variety of purposes, including diskless systems, recovery partitions and remotely overwriting other operating systems."

mailto:mm@FreeBSD.org[Martin Matuska] is the creator of mfsBSD, and he is also the author of the original link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[mfsBSD white paper], which is where the excerpt above was taken from.
The upstream mfsBSD is maintained at the link:https://github.com/mmatuska/mfsbsd[GitHub mfsBSD repository].

> ==== Purpose of this project

Reviewer: ```suggestion
==== Purpose of this Project
```

> This project integrates mfsBSD into the FreeBSD release tool set, creating an additional target of mfsBSD images (.img for disc and .iso for optical images) into the /usr/src/release/Makefile.

Reviewer: ```suggestion
This project integrates mfsBSD into the FreeBSD release tool set, creating an additional target of mfsBSD images ([.filename]#.img# for disc and [.filename]#.iso# for optical images) into the [.filename]#/usr/src/release/Makefile#.
```

> This project integrates mfsBSD into the FreeBSD release tool set, creating an additional target of mfsBSD images (.img for disc and .iso for optical images) into the /usr/src/release/Makefile.

Reviewer: For `.iso`, is _optical images_ the best phrase? 

(I imagine `.iso` files used, maybe more commonly, in other contexts.)

Author: I agree! Change commit incoming :+1:

Until now, mfsBSD only existed outside the FreeBSD release tool chain, and only the -release versions have been produced.

> With this project, mfsBSD images will be available at the official FreeBSD release page, and you will also be able to build mfsBSD yourself by invoking cd /usr/src/release && make release WITH_MFSBSD=1, which will then create mfsbsd-se.img and mfsbsd-se.iso at /usr/obj/usr/src/${ARCH}/release/.

Reviewer: ```suggestion
With this project, mfsBSD images will be available at the official FreeBSD release page, and you will also be able to build mfsBSD yourself by invoking `cd /usr/src/release && make release WITH_MFSBSD=1`, which will then create [.filename]#mfsbsd-se.img# and [.filename]#mfsbsd-se.iso# at [.filename]#/usr/obj/usr/src/${ARCH}/release/#.
```

> ==== Changes from the last quarter

Reviewer 3: Maybe

```suggestion
==== Changes from last quarter
```

The code is ready, and is currently under review.

> If you'd like to get involved with the code review process, please feel free to do so!

Reviewer 3: Please do not use contractions:

```suggestion
If you would like to get involved with the code review process, please feel free to do so!
```

> Here's link:https://reviews.freebsd.org/D41705[my code revision].

Reviewer 3: Please do not use contractions:

```suggestion
Here is link:https://reviews.freebsd.org/D41705[my code revision].
```
